agr_literature_service/api/crud/referencefile_crud.py

```diff
@@ -141,6 +141,48 @@ def file_upload(db: Session, metadata: dict, file: UploadFile):  # pragma: no co
     else:
         file_upload_single(db, metadata, file)
 
+    cleanup_temp_file(db, metadata["reference_curie"])
+
+
+def cleanup_temp_file(db: Session, ref_curie: str):  # pragma: no cover
+    if ref_curie is None:
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                            detail="The specified curie is not in the standard Alliance format and no cross "
+                            "references match the specified value.")
+    ref = db.query(ReferenceModel).filter_by(curie=ref_curie).one_or_none()
```

Comment:
the curie passed in the metadata is either a reference curie or a xref, so if ref is None we should try to read the ref through the xref. We are already doing this in the function `file_upload_single` see lines 189-196. Maybe we can move this code at the beginning of the main function (`file_upload`) and then pass the reference object directly to `file_upload_single` and `cleanup_temp_file`.

Reply:
I see. Good catch! I will move code around. @valearna

Reply:
Implemented.